evm/evm-codec/src/codec.ts

```diff
@@ -0,0 +1,20 @@
+import type { Sink } from "./sink";
+import type { Src } from "./src";
+import type { Pretty } from "./utils";
+
+export const WORD_SIZE = 32;
+
+export interface Codec<T> {
+  encode(sink: Sink, val: T): void;
+  decode(src: Src): T;
+  isDynamic: boolean;
+  slotsCount?: number;
```

Comment:
why is it needed?

Reply:
We need it to calculate the offsets for the dynamic types encoding. If it's not present, we just assume it takes 1 slot